src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -129,7 +129,7 @@ public ReadOnlyTensorSpan(ReadOnlySpan<T> span, scoped ReadOnlySpan<nint> length
             strides = strides.IsEmpty ? (ReadOnlySpan<nint>)TensorSpanHelpers.CalculateStrides(lengths, linearLength) : strides;
             TensorSpanHelpers.ValidateStrides(strides, lengths);
             nint maxElements = TensorSpanHelpers.ComputeMaxLinearIndex(strides, lengths);
-            if (maxElements >= span.Length && span.Length != 0)
+            if (span.IsEmpty ? maxElements > 0 : maxElements >= span.Length)
```

Comment:
Does `ComputeMaxLinearIndex` throw on overflow? That is, is `maxElements` guaranteed to be non-negative, in which case `maxElements != 0` might be clearer?

Reply:
`ComputeMaxLinearIndex` does not throw, but we validate and throw if a length is negative before this point, so `maxElements` is guaranteed to be non-negative.

Reply:
Fixed.